cmd/catalyst-node/catalyst-node.go

```diff
@@ -501,12 +549,19 @@ func protocol(r *http.Request) string {
 
 func queryMistForClosestNode(playbackID, lat, lon, prefix string) (string, error) {
 	// First, check to see if any server has this stream
-	_, err := queryMistForClosestNodeSource(playbackID, lat, lon, prefix, true)
-	if err != nil {
-		return "", err
-	}
-	// If so, fetch the best server for playback
-	return queryMistForClosestNodeSource(playbackID, lat, lon, prefix, false)
+	_, err1 := queryMistForClosestNodeSource(playbackID, lat, lon, prefix, true)
```

Comment:
This logic seems fishy. Shouldn't we just do like this?
1. Check the best playback server (`source = false`)
2. Only if there is no best playback server, then query for any (`source = true`)

Now, we always query for both which seems kind-of weird. If it needs to be that way, then I suggest to add a good comment which explains why it is so.

Reply:
* The `source = false` call always returns a server even if the stream isn't live anywhere. We need that server to be the 404 handler.
* The `source=true` call only returns successfully if the stream is live somewhere. We need this information to know whether we should be redirecting to `video` or `videorec`.

Don't see how we can avoid doing both, but I can add a comment to that effect